doc/site/docs/02-core-concepts/06-body.md

```diff
@@ -7,66 +7,64 @@ sidebar_label: Body
-- **Encoding support**: Handles text encoding (UTF-8, Latin-1, etc.) automatically while allowing manual override when needed
+- Stream-based architecture: All body content is handled as `Stream<Uint8List>` for memory efficiency, allowing you to process large files without loading them entirely into memory
+- One-time read constraint: Bodies can only be read once to prevent accidental double-consumption
+- Automatic type detection: Intelligent MIME type inference for JSON, HTML, XML, and binary file types
+- Encoding support: Handles text encoding (UTF-8, Latin-1, etc.) automatically with manual override options
 
 ## Creating bodies
 
 ### From string content
 
-The most common way to create a body is from string content. Relic provides intelligent automatic detection of content types based on the string's structure, though you can always override this behavior when needed.
+The most common way to create a body is from string content, and Relic can infer content types from the structure of the text you provide. This lets you write straightforward handlers while still getting correct headers and encodings without extra work.
 
-For simple text content, you can create a body directly from a string:
+For simple text content, create a body directly from a string and return it as part of your response. Relic will handle encoding and set a sensible default content type for you:
 
 ```dart
 final body = Body.fromString('Hello, World!');
 ```
 
-When you provide JSON content, Relic automatically detects the MIME type by analyzing the string structure:
+For JSON content, Relic detects the MIME type based on the shape of the string. You can return JSON without manually setting headers, which keeps handlers concise and clear:
 
 ```dart
 final jsonBody = Body.fromString('{"message": "Hello"}');
-// Automatically detects application/json MIME type
+// Automatically detects application/json
 ```
 
-Similarly, HTML content is automatically recognized and tagged with the appropriate MIME type:
+HTML can be treated the same way, with automatic detection ensuring that browsers and clients receive the correct content type. This reduces the need to repeat metadata that Relic can determine on your behalf:
 
 ```dart
 final htmlBody = Body.fromString('<!DOCTYPE html><html>...</html>');
-// Automatically detects text/html MIME type
+// Automatically detects text/html
 ```
 
-For cases where you need explicit control over the MIME type and encoding, you can specify these parameters directly:
+When you need explicit control, specify the MIME type and encoding directly. This is useful for legacy encodings, strict APIs, or any time you want to override detection for clarity or compliance:
 
 ```dart
 final customBody = Body.fromString(
-  'Custom content',
+  'Héllo world',
   mimeType: MimeType.plainText,
   encoding: latin1,
 );
 ```
 
-### From binary data
+### From files
```

Comment:
We should add an "info" box somewhere here that mentions the static file handler and that is the way to serve a directory of static files.

Reply:
Added